### PR TITLE
LPS-177382 Remove unused interface com.liferay.portal.search.elasticsearch7.settings.IndexSettingsContributor and its implementation

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
Force_Push is just to rebase and  resolve the conflict about file  `modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-api/bnd.bnd`

**Comment by W2jx** [comment]
ci:test:search

**Comment by W2jx** [comment]
ci:test:sf

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
no conflicts(3/7 9:02)

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/index/CompanyIndexFactoryTest.java [hunk 1]
```diff
@@ -157,26 +138,6 @@ public void testAdditionalTypeMappingsWithRootType() throws Exception {
 		_assertAdditionalTypeMappings();
 	}
 
-	@Test
-	public void testAdditionalTypeMappingsWithRootTypeFromContributor()
```
Comment (peerkar) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/test/java/com/liferay/portal/search/elasticsearch7/internal/index/CompanyIndexFactoryTest.java:161]: @lipusz FYI that the SPI version is not covered by this one test, but we shouldn't be even needing setting or changing the ' root type' any more so I guess it's acceptable?

**Comment by peerkar** [comment]
@W2jx looks ok to me to go.

**Comment by peerkar** [comment]
ci:forward

**Pull request closed without merging**